### PR TITLE
Resolve missing gnuplot plots in manual

### LINKED ISSUE #1191: How to compile the “tex” files to generate the documentation pgfmanual.pdf
### Version

_No response_

### Details

I want to compile the “tex” files in ``` .\doc\generic\pgf``` to generate pgfmanual.pdf. But no matter whicn compiler I used，it always carry out fault。
Is there a course to generate documentation using the files in ```.\doc```

### LINKED ISSUE #1238: gnuplot plots are all missing
### Brief outline of the bug

As uncovered in https://github.com/pgf-tikz/pgf/issues/1237#issuecomment-1426873045, in building pgfmanual v3.10.0 all pre-generated gnuplots data files are not found, hence all `\draw plot[...] file {plots/xxx.table};` examples mainly in sec. _Plotting a Function Using Gnuplot_ plot nothing.

pgfmanual v3.9.1a doesn't have this problem.

### Minimal working example (MWE)

not required

### PR DESCRIPTION
**Pull request by muzimuzhi**
**Motivation for this change**

The last commit is just temp and should be removed before merging.

Fixes #1191
Fixes #1238

**Checklist**

Please [signoff your commits][git-s] to explicitly state your agreement to the [Developer Certificate of Origin][DCO]. If that is not possible you may check the boxes below instead:

- [x] Code changes are licensed under [GPLv2][GPL] + [LPPLv1.3c][LPPL]
- [x] Documentation changes are licensed under [FDLv1.2][FDL]

[git-s]: https://git-scm.com/docs/git-commit#Documentation/git-commit.txt--s
[DCO]: https://developercertificate.org
[GPL]: https://www.gnu.org/licenses/gpl-2.0.html
[LPPL]: https://www.latex-project.org/lppl/lppl-1-3c.txt
[FDL]: https://www.gnu.org/licenses/fdl-1.2.html

### CODE REVIEW

File: build.lua [hunk 1]
```diff
@@ -22,6 +22,10 @@ flatten = false
 -- To allow writing
 function docinit_hook()
   mkdir(typesetdir .. "/plots")
+  -- Execute cp() manually to keep the directory structure, otherwise if
+  -- "plots/*.table" is appended to docfiles, ".table" files will be copied
+  -- directly to typesetdir, not typesetdir .. "/plots".
+  cp("plots/*.table", docfiledir, typesetdir .. "/plots")
```
Comment (muzimuzhi) [inline build.lua]: To keep treatments unified, do similar special treatment for `"images/*.jpg"`?

Comment (muzimuzhi) [reply build.lua]: Alternatively, appending the subdir name to `docfiles` does the trick. @josephwright Is this the expected behavior or a misused "feature"?
```diff
diff --git a/build.lua b/build.lua
index 7bf4f678..9e712ddb 100644
--- a/build.lua
+++ b/build.lua
@@ -12,7 +12,7 @@ docfiledir = "./doc/generic/pgf"
 docfiles =
   {
     "RELEASE_NOTES.md", "description.html", -- Part of the release script
-    "color.cfg", "pgfmanual.cfg", "images/*.jpg", "*.tex" -- Build the PDF
+    "color.cfg", "pgfmanual.cfg", "images", "plots", "*.tex" -- Build the PDF
   }
 tdsroot = "generic"
 typesetfiles = {"pgfmanual.tex"}
```

**Comment by muzimuzhi** [comment]
> The last commit is just temp and should be removed before merging.

`\usetikzpackage` raises error for unknown package name. I'm now leaning towards (officially) promote "Plot data file not found" from warning to error.

File: build.lua (hunk 1 above)
Comment (hmenke) [reply build.lua]: @josephwright ping

Comment (muzimuzhi) [reply build.lua]: Another idea: if `\input@path`/`\l_file_search_path_seq` is respected by `\pgfutil@IfFileExists` and `\pgfutil@InputIfFileExists` (see #1259), then we can drop prefix `plots/` in manual examples
https://github.com/pgf-tikz/pgf/blob/14ef7f3494766f3df4c9b177bec1e5695ec980f5/doc/generic/pgf/pgfmanual-en-tikz-plots.tex#L146

**Review by hmenke — APPROVED** [review]

File: build.lua (hunk 1 above)
Comment (muzimuzhi) [reply build.lua]: > Alternatively, appending the subdir name to `docfiles` does the trick. @josephwright Is this the expected behavior or a misused "feature"?

From conversation in https://github.com/latex3/l3build/issues/331 and a casual check in `l3build` source, yes it is supported. I tend to use this currently undocumented feature in pgf's `l3build` configuration.

As this PR also promotes missing plot data file from warning to error, feature breaking change in `l3build` will be caught.

**Comment by muzimuzhi** [comment]
As https://github.com/pgf-tikz/pgf/commit/d1ada29d65a94acc958ad3340a918c0c64534c07 didn't pass the "Check commits" check, I reorganized commits and reworded CHANGELOG entries (perhaps overly), and force pushed again (sorry).

**Review by hmenke — APPROVED** [review]
Perfect :ok_hand:

**Pull request merged**